docs_src/design/adr/core/0026-Core-Data-Cache.md

```diff
@@ -0,0 +1,85 @@
+    - N must be > 0
+
+### Delete operations (to keep the number of readings to the "latest" or a database size limit)
+
+Core data will add the following new delete operations.  Note that only one will be exposed externally via API.  The other will be used internally by the service (when configured to use it).
+
+- Delete all but N readings (aka - the **Eaton Delete**; clean readings but preserve the latest readings)
+    - API:  *DELETE* /reading/keep/{N}
+    - if there are fewer than (or equal to) N readings in core data, then all readings would be kept and no readings would be removed
+    - if there are more than N readings in core data, then delete the oldest readings, but keep the youngest N readings
+    - N must be > 0 (N = 1 means keep only the latest reading)
+    - this delete option will be triggered via API either:
+        - manually by the adopter
+        - or by the scheduling service (the adopter would set up a new IntervalAction with the scheduling service)
+
+- Delete readings to keep the database at a certain size (aka - the **Tony Delete**; clean readings to preserve the number of readings in the database)
+    - API:  none.  This operation is not visible via REST API
+    - this operation, when called upon internally by core data, will remove readings (from oldest to newest) to keep the core data database within a specified size
+    - a configuration setting (`CacheDelete` in configuration) will determine how readings will be removed by this operation:
+        - option A (`CacheDelete="OnEach"`): remove one reading (the oldest reading) each time a new reading gets added
+            - this operation would be called as part of the core data add event/reading operations
+            - this option causes core data to operate like a real cache - keeping only the latest reading
+        - option B (`CacheDelete="OnCount"`): remove a specified number (`DeleteSize` in configuration) of readings when the number of readings exceeds a specified limit (`CacheSize` in configuration)
+            - this operation would be called as part of the core data add event/reading operations
+            - this option causes core data to keep within a size range (between a max and min number of readings) 
+            - examples: 
+                - if there are 100 readings in the database and the `CacheSize` of the database readings is set to 100, then when the next reading is added to core data, a delete of the latest `DeleteSize` readings are removed (starting with the oldest readings) from core data.
+                - if there are 80 readings in the database and the `CacheSize` of the database readings is set to 100, then when the next reading is added to core data, no delete occurs.  The reading gets added normally and the number of readings in the database is now 81.
+            - this option reduces the add/delete thrashing.  It only deletes when the max size is hit or exceeded
+
```

Comment:
I found the Redis areas (maxmemory and its policy) that are built-in for control of size:
https://redis.io/docs/manual/config/#configuring-redis-as-a-cache
https://redis.io/docs/manual/eviction/
*As long as* the event redis db was separate from the core-metadata redis db, could this be all that we need for the "Tony" solution? It's blind to what data that is dropped, but is an existing cudgel that we might choose to use.

Reply:
using the database mechanism is a good option, but you might be hard to control which readings you want to keep.